pkg/daytona/unmarshal.go

```diff
@@ -0,0 +1,251 @@
+package daytona
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/api"
+)
+
+const (
+	vaultTagPath = "vault_path"
+	vaultTagKey  = "vault_key"
+
+	vaultTagPathKey = "vault_path_key"
+
+	defaultVaultTagField = "value"
+)
+
+var ErrValueInput = errors.New("the provided value must be a struct pointer")
+
+type unmarshalSecretsConfig struct {
+	apex string
+}
+
+// An UnmarshalSecretsOption is an option for the UnmarshalSecrets function
+type UnmarshalSecretsOption interface {
```

Comment:
Instead of doing an interface here could we do the normal options pattern by using a `type Option func(*unmarshalSecretsConfig)`   then we don't need to have `type withApex string`  we can just have the function 
```go
func WithApex(apex string) Option {
  return func(c *unmarshalSecretsConfig) {
    c.apex = apex
  }
}